src/webui/www/private/scripts/client.js

```diff
@@ -928,6 +928,25 @@ window.addEventListener("DOMContentLoaded", () => {
 
         $("freeSpaceOnDisk").textContent = "QBT_TR(Free space: %1)QBT_TR[CONTEXT=HttpServer]".replace("%1", window.qBittorrent.Misc.friendlyUnit(serverState.free_space_on_disk));
 
+        if (window.qBittorrent.Cache.preferences.get().status_bar_external_ip) {
+            const lastExternalAddressV4 = serverState.last_external_address_v4;
+            const lastExternalAddressV6 = serverState.last_external_address_v6;
+            const hasIPv4Address = lastExternalAddressV4 !== "";
+            const hasIPv6Address = lastExternalAddressV6 !== "";
+            let lastExternalAddressLabel = "QBT_TR(External IP: N/A)QBT_TR[CONTEXT=HttpServer]";
+            if (hasIPv4Address && hasIPv6Address)
+                lastExternalAddressLabel = "QBT_TR(External IPs: %1, %2)QBT_TR[CONTEXT=HttpServer]";
+            else if (hasIPv4Address || hasIPv6Address)
+                lastExternalAddressLabel = "QBT_TR(External IP: %1%2)QBT_TR[CONTEXT=HttpServer]";
+            // replace in reverse order ('%2' before '%1') in case address contains a % character.
+            // for example, see https://en.wikipedia.org/wiki/IPv6_address#Scoped_literal_IPv6_addresses_(with_zone_index)
+            $("externalIPs").textContent = lastExternalAddressLabel.replace("%2", lastExternalAddressV6).replace("%1", lastExternalAddressV4);
+
+            const elem = document.getElementById("externalIPs");
+            elem.classList.remove("invisible");
+            elem.previousElementSibling.classList.remove("invisible");
```

Comment:
Shouldn't you have a matching `else` case? to hide the ui element on-the-fly.

Reply:
It's not needed because the page is reloaded any time the preferences are updated. Any we never re-fetch preferences after the initial page load.

Reply:
>It's not needed because the page is reloaded any time the preferences are updated. 

In the future, I would like to not reload the webui unless necessary. So for options that doesn't require a reload, the aforementioned `else` case would be required.